framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/ASceneWrapper.java

```diff
@@ -0,0 +1,251 @@
+                removeIgnoredAnnosFromATypeElement(
+                        field.type, TypeUseLocation.FIELD, annosToIgnore);
+            }
+            for (AMethod method : aclass.methods.values()) {
+                // Return type
+                removeIgnoredAnnosFromATypeElement(
+                        method.returnType, TypeUseLocation.RETURN, annosToIgnore);
+                // Receiver type
+                removeIgnoredAnnosFromATypeElement(
+                        method.receiver.type, TypeUseLocation.RECEIVER, annosToIgnore);
+                // Parameter types
+                for (AField param : method.parameters.values()) {
+                    removeIgnoredAnnosFromATypeElement(
+                            param.type, TypeUseLocation.PARAMETER, annosToIgnore);
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes all annotations that should be ignored from an ATypeElement.
+     *
+     * @param typeElt the type element from which to remove annotations
+     * @param loc the location where typeEl in used
+     * @param annosToIgnore maps a pair of
+     *     <ul>
+     *       <li>the toString() representation of an ATypeElement's description concatenated with
+     *           its annotations
+     *       <li>the ATypeElement's TypeUseLocation
+     *     </ul>
+     *     to a set of names of annotations that should not be added to .jaif files for that
+     *     location.
+     */
+    private void removeIgnoredAnnosFromATypeElement(
+            ATypeElement typeElt,
+            TypeUseLocation loc,
+            Map<Pair<String, TypeUseLocation>, Set<String>> annosToIgnore) {
+        String annosToIgnoreKey = typeElt.description.toString() + typeElt.tlAnnotationsHere;
+        Set<String> annosToIgnoreForLocation = annosToIgnore.get(Pair.of(annosToIgnoreKey, loc));
+        if (annosToIgnoreForLocation != null) {
+            Set<Annotation> annosToRemove = new HashSet<>();
+            for (Annotation anno : typeElt.tlAnnotationsHere) {
+                if (annosToIgnoreForLocation.contains(anno.def().toString())) {
+                    annosToRemove.add(anno);
+                }
+            }
+            typeElt.tlAnnotationsHere.removeAll(annosToRemove);
+        }
+
+        // Recursively remove ignored annotations from inner types
+        for (ATypeElement innerType : typeElt.innerTypes.values()) {
+            removeIgnoredAnnosFromATypeElement(innerType, loc, annosToIgnore);
+        }
+    }
+
+    /**
+     * Write the scene wrapped by this object to a .jaif file on the given path.
+     *
+     * @param jaifPath the path to the jaif file to be written
+     * @param annosToIgnore which annotations should be ignored in which contexts
+     */
+    public void writeToJaif(
+            String jaifPath, Map<Pair<String, TypeUseLocation>, Set<String>> annosToIgnore) {
+        removeIgnoredAnnosFromScene(theScene, annosToIgnore);
+        theScene.prune();
```

Comment:
Either undo this, or add to the documentation that the method side-effects the scene.  Same comment for writeToStub.